plan-for-extension-exercise/src/planForExtensionExercise/TotalAreaCalculator.java

```diff
@@ -0,0 +1,25 @@
+package planForExtensionExercise;
+
+public class TotalAreaCalculator extends AreaCalculator {
+	public static double calculateArea(Shape[] shapes) {
+		double area = 0.0;
+		if (shapes == null) {
+			return area;
```

Comment:
That's certainly valid, however could be misleading. Is area of a null 0? Maybe yes...what if you truly want the user to know that what was passed in is in fact an unexpected parameter (like null). What would you do without changing the return type, however (hint) it may require a method signature to be changed, and how?